internal/types/types.go

```diff
@@ -1,7 +1,14 @@
 package types
 
+import "time"
+
 // Logger is a subset of log.Logger to allow compatible loggers to be used with Sync.
 type Logger interface {
 	Println(v ...any)
 	Printf(format string, v ...any)
 }
+
+// Clock is a subset of time.Time which allows us to mock the clock in tests.
+type Clock interface {
+	Now() time.Time
+}
```

Comment:
Instead of putting this in the types folder, can you move this to co-locate it with your OnCall code. We can centralise it if we need it in subsequent adapters.